Formula/synergy-core.rb

```diff
@@ -0,0 +1,351 @@
+           "  Synergy may not work after being upgraded.\n\n" \
+           "  It is not adequate to untick the checkbox beside each relevant application. You must\n" \
+           "  remove each application from the list using the '-' button before re-adding them.\n\n" \
+           "  Alternatively, run this command to clear the 'Accessibility' permissions page:\n" \
+           "    sudo tccutil reset Accessibility\n" \
+           "  After running this command, you will need to re-add programs that you want in the list.\n\n" \
+           "  Make sure that you exit Synergy before revoking the 'Accessibility' permission. If\n" \
+           "  Synergy is running when the permission is revoked, you may experience highly anomalous\n" \
+           "  behavior with your mouse and keyboard."
+      if Dir.exist?("/Applications/Synergy.app")
+        s += "\n\n" \
+             "If you previously installed Synergy without using this formula, you may want to\n" \
+             "delete your previous installation to avoid confusion. It appears that you have a\n" \
+             "previous non-brew installation of Synergy at /Applications/Synergy.app. You may\n" \
+             "experience problems granting the 'Accessibility' permission to both copies of\n" \
+             "Synergy at once, even if they aren't running at the same time."
+      end
+      return s
+    end
+    on_linux do
+      return "On Linux, Synergy does not support Wayland. You must use Xorg.\n" \
+             "For more information, see:\n" \
+             "  https://github.com/symless/synergy-core/issues/4090"
+    end
+  end
+
+  # We'll check for these messages in the tests below.
+  LOG_PERMISSION_ERROR = "FATAL: A runtime error occurred: " \
+                         "assistive devices does not trust this process, allow it in system settings.".freeze
+  LOG_NO_DISPLAY_ERROR = "FATAL: A display invalid exception error occurred: failed to initialize screen shape".freeze
+
+  test do
+    port = free_port
+    conf_file = testpath/"synergy.conf"
+    conf_file.write <<~EOS
+      section: screens
+        test-server:
+        test-client:
+      end
+    EOS
+
+    # The Synergy client and server print messages with two formats:
+    #   [2022-04-24T15:18:27] NOTE: started client
+    # and
+    #    2022-04-24 15:18:27.816 synergyc[57673:302843] starting cocoa loop
+    #
+    # In both cases, the message text starts after the first ']' character.
+    log_trim = ->(log_line) { log_line.split("]", 2)[1]&.strip! }
+
+    # In certain cases, only limited testing is possible. In such cases, we will
+    # print this message.
+    def ohai_nontrivial_testing_is_not_possible
+      ohai "Thus, it is not possible to do any nontrivial testing of synergy-core."
+      ohai ""
+      ohai "This is not considered a test failure because it does not indicate any problem"
+      ohai "with brew's installation of synergy-core."
+    end
```

Comment:
We don't want to print anything in tests

Reply:
I've removed almost all of the printing. However, I've retained a few messages that I believe are appropriate and consistent with brew style:
- First, I retained a message if the majority of the tests cannot be run because of insufficient permissions or lack of an attached display. I believe the user should know that many tests were skipped -- and more importantly, they should know what they have to do if they want those tests to run. I made the messages a lot shorter and simpler though.
- Second, I retained a warning message that the user's mouse and keyboard may behave strangely during the tests. I believe the user could be alarmed by this behavior without the warning. Some users might even (incorrectly) associate such anomalies with the presence of malicious code (see https://xkcd.com/1180/).
- Third, I retained the printing of the output of the programs run by the tests. I did this for the following reason: By default, `brew test` prints program output when programs are invoked using `system`. Since we invoke these programs using `IO.popen` in order to have greater control over streams, brew is unable to print their output to the screen without our help. To preserve this default behavior, we print the program output to the console ourselves. I added a comment with this explanation.